.github/workflows/storybook-deploy.yml

```diff
@@ -0,0 +1,40 @@
+name: Build and Deploy Storybook
+
+on:
+  push:
+    branches:
+      - main
+  workflow_call:
+    secrets:
+      PUBLISH_STORYBOOK_TOKEN:
+        required: true
+
+jobs:
+  build-deploy-storybook:
+    name: Build and Deploy Storybook
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    environment: github-pages
+    defaults:
+      run:
+        working-directory: apps/storybook-react
+    steps:
+      - name: Checkout and setup environment
+        uses: MetaMask/action-checkout-and-setup@v1
+        with:
+          is-high-risk-environment: false
+          cache-node-modules: true
```

Comment:
Not sure if there is an alternative here to using `cache-node-modules`. Maybe @itsyoboieltr knows?